src/XMakeBuildEngine/Construction/ProjectElementContainer.cs

```diff
@@ -619,7 +633,11 @@ private void AddInitialChild(ProjectElement child)
             AddToXml(child);
 
             _count++;
-            MarkDirty("Add child element named '{0}'", child.ElementName);
+
+            if (HasXmlRepresentation(child))
```

Comment:
Wouldn't you also want to suppress marking the tree as dirty if the child is implicit in `InsertBeforeChild` and `InsertAfterChild`?